ofrak_core/CHANGELOG.md

```diff
@@ -0,0 +1,53 @@
+# Changelog
+All notable changes to `ofrak` will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) and adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased](https://github.com/redballoonsecurity/ofrak/tree/master)
+
+## [2.0.0](https://github.com/redballoonsecurity/ofrak/releases/tag/ofrak-v2.0.0) - 2023-01-03
+### Added
+- `ofrak` now contains all the components that were in the (now sunset) `ofrak-components` package.
+- Add components for analyzing, unpacking, and repacking raw flash dumps.
+- UF2 unpacker and packer added.
+- Several new features to make tracking and installing external dependencies easier (for non-Docker users):
+  - Components can now mark their external tool dependencies using the `external_dependencies` attribute.
+  - CLI tool added to track and install dependencies & components:
+    - `python -m ofrak list` lists installed OFRAK modules and/or components.
+    - `python -m ofrak deps` shows and checks external (non-Python) dependencies of OFRAK components, and shows hints on how they might be installed.
+  - `OFRAKContext` logs if a discovered component is missing an external dependency; these components can be excluded using the `exclude_components_missing_depenencies` parameter in `OFRAK`.
+- Speed improvements to `resource.unpack_recursively` method.
+- `DataService` was refactored for speed and clarity of implementation.
+
+### Changed
+- `Resource.{write_to, flush_to_disk}` now have optional `pack` parameter to pack before writing
+- Resource views are now updated when `Resource.save` is run
+- Free space components decoupled from ISA
+- `Filesystem.initialize_from_disk` raises `FileNotFoundError` when path does not exist or is not a file ([@galbwe](https://github.com/galbwe))
+- Rename `RawExtendModifier` to `BinaryExtendModifier`
```

Comment:
These should probably end with periods for consistency with the rest of this document.

Reply:
Worth doing a pass over all the bullets to make sure they're consistent – there are a few other places lower in the document that are also missing consistent trailing punctuation.

Reply:
Good catch, I went through all the docs again as well.